### ISSUE TITLE
[5/6/7] -Wunused-const-variable false positive for variable template

### ISSUE THREAD

**Issue by CaseyCarter** [llvm/llvm-project#37851]
|  |  |
| --- | --- |
| Bugzilla Link | [38503](https://llvm.org/bz38503) |
| Version | trunk |
| OS | Windows NT |

## Extended Description 
Every clang since 5.0 emits -Wunused-const-variable for a TU that contains only:

  template <class> constexpr bool always_false = false;

Note that there are no variables here, only a variable template.

**Comment by wheatman** [comment]
This was fixed between 8.0.1 and 9.0.0
https://godbolt.org/z/YeEvdEb5T

code, compile with `-Wunused-const-variable`
```
 template <class> constexpr bool always_false = false;
```

old warning
```
<source>:1:34: warning: unused variable 'always_false' [-Wunused-const-variable]
 template <class> constexpr bool always_false = false;
                                 ^
1 warning generated.
Compiler returned: 0
```
